api/src/opentrons/protocol_engine/types.py

```
@@ -837,3 +837,60 @@ def from_hw_state(cls, state: HwTipStateType) -> "TipPresenceStatus":
             HwTipStateType.PRESENT: TipPresenceStatus.PRESENT,
             HwTipStateType.ABSENT: TipPresenceStatus.ABSENT,
         }[state]
+
+
+class RTPBase(BaseModel):
+    """Parameters defined in a protocol."""
+
+    displayLabel: str = Field(..., description="Display string for the parameter.")
+    variableName: str = Field(..., description="Python variable name of the parameter.")
+    description: str = Field(..., description="Detailed description of the parameter.")
+    suffix: Optional[str] = Field(
+        None,
+        description="Units (like mL, mm/sec, etc) or a custom suffix for the parameter.",
+    )
+
+
+class IntParameter(RTPBase):
+    """An integer parameter defined in a protocol."""
+
+    min: int = Field(
+        ..., description="Minimum value that the integer param is allowed to have."
+    )
+    max: int = Field(
+        ..., description="Maximum value that the integer param is allowed to have."
+    )
+    default: int = Field(
+        ...,
+        description="Default value of the parameter, to be used when there is no client-specified value.",
+    )
```

Beware that Pydantic has very nasty type coercion behavior by default. When parsing an `int` field, it will silently truncate a JSON `3.14159` value into `3`.

I think the way to fix this in our version Pydantic, if you want to, is `pydantic.StrictInt`.

---

I just remembered that we've refrained from using `StrictInt` in fields that are similar to this one where the expected type on the backend is an integer. We've mostly left the fields be `floats` with purposeful validation/ conversion down the line. I don't remember the exact reasons (maybe you or @shlokamin know why) but that actually makes things simpler and more straightforward in this case. 

 If we only have a `float` param model (we will still have different int & float param API functions) and rely on the engine-based param validator and app's validation (which we will implement anyway), then we won't run into weird pydantic coercions.

---

The other option is to keep the two different models and make them strict type so that `Union[intParam, floatParam]` won't try to coerce one type into other, but it feels a bit unnecessary since we will actually have float vs int validation on both ends of this client-server communication.

Let me know if I'm missing something though.

---

Sure, whatever is easiest for you all.

---

Cool, I'm going to merge this as is once it gets an approval since I am blocked on it, but will make a change in the follow-up. It's not a critical issue and I'm sure we'll have a few more changes to this.